### PR TITLE
Add web and list-port commands to mzcompose

### PR DESCRIPTION
**Pull request by cirego**
These helpers commands from mzconduct and are simple ports from their
previous implementations. They are useful for determining the high
numbered ports that Docker Compose allocates for services when they are
not bound to well-known ports.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5075)

### CODE REVIEW

File: misc/python/materialize/cli/mzcompose.py [hunk 1]
```diff
@@ -149,6 +169,24 @@ def gen_shortcuts(repo: mzbuild.Repository) -> int:
...
+    containers = composition.run(["ps", "-q"], capture=True).stdout.splitlines()
+    metadata = spawn.capture(["docker", "inspect", "-f", "{{json .}}", *containers,])
+    metadata = [json.loads(line) for line in metadata.splitlines()]
+    ports = []
+    for md in metadata:
+        if md["Config"]["Labels"]["com.docker.compose.service"] == service:
+            for (name, port_entry) in md["NetworkSettings"]["Ports"].items():
+                for p in port_entry or []:
+                    ports.append(p["HostPort"])
+    return ports
```
Comment (benesch) [inline misc/python/materialize/cli/mzcompose.py:187]: This seems like it wants to be a method on `Composition`.

**Review by benesch — APPROVED** [review]
Thanks!

File: misc/python/materialize/cli/mzcompose.py (hunk 1 above)
Comment (cirego) [reply misc/python/materialize/cli/mzcompose.py:187]: Can do!

Comment (cirego) [reply misc/python/materialize/cli/mzcompose.py]: Turns out it already was!

**Pull request merged**